### PR TITLE
Document 1.0.0 migration nuances

### PR DESCRIPTION
**Pull request by levan-m**
### What does this PR do?

Adding `0.8.x -> 1.0.0` migration nuances to the readme.

### Motivation

https://github.com/DataDog/datadog-operator/issues/754

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,15 @@ However, the Datadog Operator is still in beta, so it is not yet a recommended w
 
 See the [Getting Started][5] dedicated documentation to learn how to deploy the Datadog operator and your first Agent, and [Configuration][12] to see examples, a list of all configuration keys, and default values.
 
+### Migrating from `0.8.x` to `1.0.0`
+
+Operator `1.0.0` contains several changes users need to be aware of:
+* `DatadogAgent` CRD has two versions, `v1alpha1` and `v2alpha1`. They are used as a stored version by Operator `0.8.x` and `1.0.0` respectively. Please see this [Kubernetes documentation page][13] for more details about CRD versioning.
+* `v1alpha1` and `v2alpha1` are not backward or forward compatible. The Datadog Operator `1.0.0` implements a Conversion Webhook to migrate, though it only supports from `v1alpha1` to `v2alpha1`.
+* With the Conversion Webhook enabled users can ran `1.0.0` but continue applying `v1alpha1` manifest. However, they won't be able to retrieve the `DatadogAgent` manifest as a `v1alpha1` object (see the previous item).
```
Comment (CharlyF) [inline README.md:36]:
```suggestion
* With the Conversion Webhook enabled users can run `1.0.0` but continue applying `v1alpha1` manifest. However, they won't be able to retrieve the `DatadogAgent` manifest as a `v1alpha1` object (see the previous item).
```

**Review by CharlyF — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -28,6 +28,15 @@ However, the Datadog Operator is still in beta, so it is not yet a recommended w
 
 See the [Getting Started][5] dedicated documentation to learn how to deploy the Datadog operator and your first Agent, and [Configuration][12] to see examples, a list of all configuration keys, and default values.
 
+### Migrating from `0.8.x` to `1.0.0`
+
+Operator `1.0.0` contains several changes users need to be aware of:
+* `DatadogAgent` CRD has two versions, `v1alpha1` and `v2alpha1`. They are used as a stored version by Operator `0.8.x` and `1.0.0` respectively. Please see this [Kubernetes documentation page][13] for more details about CRD versioning.
+* `v1alpha1` and `v2alpha1` are not backward or forward compatible. The Datadog Operator `1.0.0` implements a Conversion Webhook to migrate, though it only supports from `v1alpha1` to `v2alpha1`.
+* With the Conversion Webhook enabled users can run `1.0.0` but continue applying `v1alpha1` manifest. However, they won't be able to retrieve the `DatadogAgent` manifest as a `v1alpha1` object (see the previous item).
```
Comment (hestonhoffman) [inline README.md:36]:
```suggestion
* With the Conversion Webhook enabled, users can run `1.0.0` but continue applying a `v1alpha1` manifest. However, they won't be able to retrieve the `DatadogAgent` manifest as a `v1alpha1` object (see the previous item).
```

File: README.md [hunk 3]
```diff
@@ -28,6 +28,15 @@ However, the Datadog Operator is still in beta, so it is not yet a recommended w
 
 See the [Getting Started][5] dedicated documentation to learn how to deploy the Datadog operator and your first Agent, and [Configuration][12] to see examples, a list of all configuration keys, and default values.
 
+### Migrating from `0.8.x` to `1.0.0`
+
+Operator `1.0.0` contains several changes users need to be aware of:
+* `DatadogAgent` CRD has two versions, `v1alpha1` and `v2alpha1`. They are used as a stored version by Operator `0.8.x` and `1.0.0` respectively. Please see this [Kubernetes documentation page][13] for more details about CRD versioning.
+* `v1alpha1` and `v2alpha1` are not backward or forward compatible. The Datadog Operator `1.0.0` implements a Conversion Webhook to migrate, though it only supports from `v1alpha1` to `v2alpha1`.
```
Comment (hestonhoffman) [inline README.md:35]:
```suggestion
* `v1alpha1` and `v2alpha1` are not backward or forward compatible. The Datadog Operator `1.0.0` implements a Conversion Webhook to migrate, though it only supports migrating from `v1alpha1` to `v2alpha1`.
```

File: README.md [hunk 4]
```diff
@@ -28,6 +28,15 @@ However, the Datadog Operator is still in beta, so it is not yet a recommended w
...
+* With the Conversion Webhook enabled users can run `1.0.0` but continue applying `v1alpha1` manifest. However, they won't be able to retrieve the `DatadogAgent` manifest as a `v1alpha1` object (see the previous item).
+*  The Conversion Webhook requires a cert manager, please see the migration guide in the [public][14] or [helm chart][15] documentation for more details.
```
Comment (hestonhoffman) [inline README.md:37]:
```suggestion
*  The Conversion Webhook requires a cert manager. See the migration guide in the [public][14] or [helm chart][15] documentation for more details.
```

File: README.md [hunk 5]
```diff
@@ -28,6 +28,15 @@ However, the Datadog Operator is still in beta, so it is not yet a recommended w
 
 See the [Getting Started][5] dedicated documentation to learn how to deploy the Datadog operator and your first Agent, and [Configuration][12] to see examples, a list of all configuration keys, and default values.
 
+### Migrating from `0.8.x` to `1.0.0`
+
+Operator `1.0.0` contains several changes users need to be aware of:
+* `DatadogAgent` CRD has two versions, `v1alpha1` and `v2alpha1`. They are used as a stored version by Operator `0.8.x` and `1.0.0` respectively. Please see this [Kubernetes documentation page][13] for more details about CRD versioning.
```
Comment (hestonhoffman) [inline README.md:34]:
```suggestion
* `DatadogAgent` CRD has two versions, `v1alpha1` and `v2alpha1`. They are used as a stored version by Operator `0.8.x` and `1.0.0` respectively. See this [Kubernetes documentation page][13] for more details about CRD versioning.
```

**Review by hestonhoffman — COMMENTED** [review]
Made some edits

**Pull request merged**